src/Drivers/SEP/SEP.php

```diff
@@ -154,6 +154,10 @@ public function verify(): ReceiptInterface
             $this->notVerified($responseData['ResultCode']);
         }
 
+        if ($this->invoice->getTransactionId() !== Request::input('ResNum')) {
```

Comment:
move this line to line 133 before `$this->client->post` to fail fast before calling the remote back API.

Reply:
also by adding this line, it doesn't prevent the double payment! BTW the double payment can happen again and since each time the RefNum will be the same, it is not count as a `vulnerability` and can be catched by the merchant.